app/src/main/java/com/babylon/wallet/android/presentation/model/Amount.kt

```diff
@@ -6,99 +6,63 @@ import kotlinx.serialization.Contextual
 import kotlinx.serialization.Serializable
 import rdx.works.core.domain.resources.Resource
 
-sealed interface Amount // TODO might not needed
+sealed interface Amount
+
+data class NonFungibleAmount(
+    val certain: List<Resource.NonFungibleResource.Item>,
+    val additional: CountedAmount? = null
```

Comment:
I thought about designing the amount like this, but we don't currently have a case for `NonFungibleAmount` with `additional` being `CountedAmount.Predicted`.

Reply:
On the other hand, not needing to have a separate type for non fungible amounts when it is certain only seems to be better.

Reply:
Right, but it's better to somehow avoid duplicating these. The alternative would be to treat the Predicted type separately from the others, but it comes with other trade-offs.

Reply:
Yes exactly I was thinking about the same. As you say that would require bigger effort.